----------------------------------------------------------------------------------------------------
10:09:42 AM
username not found

   at FirmWebApiDemo.Authentication.ValidateUser.Login(String username, String password) in F:\prajval-gahine\prajval-gahine\04.WebApi\Intro to web api\FirmWebApiDemo\FirmWebApiDemo\Authentication\ValidateUser.cs:line 46
   at FirmWebApiDemo.Authentication.BasicAuthentication.OnAuthorization(HttpActionContext actionContext) in F:\prajval-gahine\prajval-gahine\04.WebApi\Intro to web api\FirmWebApiDemo\FirmWebApiDemo\Authentication\BasicAuthentication.cs:line 48
   at System.Web.Http.Filters.AuthorizationFilterAttribute.OnAuthorizationAsync(HttpActionContext actionContext, CancellationToken cancellationToken)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.AuthorizationFilterAttribute.<ExecuteAuthorizationFilterAsyncCore>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Controllers.ExceptionFilterResult.<ExecuteAsync>d__6.MoveNext()

----------------------------------------------------------------------------------------------------
----------------------------------------------------------------------------------------------------
11:45:02 AM
Object reference not set to an instance of an object.

   at FirmWebApiDemo.Authentication.BearerAuthenticationAttribute.OnAuthorization(HttpActionContext actionContext) in F:\prajval-gahine\prajval-gahine\04.WebApi\Intro to web api\FirmWebApiDemo\FirmWebApiDemo\Authentication\BearerAuthenticationAttribute.cs:line 29
   at System.Web.Http.Filters.AuthorizationFilterAttribute.OnAuthorizationAsync(HttpActionContext actionContext, CancellationToken cancellationToken)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.AuthorizationFilterAttribute.<ExecuteAuthorizationFilterAsyncCore>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Controllers.ExceptionFilterResult.<ExecuteAsync>d__6.MoveNext()

----------------------------------------------------------------------------------------------------
----------------------------------------------------------------------------------------------------
11:46:21 AM
Object reference not set to an instance of an object.

   at FirmWebApiDemo.Authentication.BearerAuthenticationAttribute.OnAuthorization(HttpActionContext actionContext) in F:\prajval-gahine\prajval-gahine\04.WebApi\Intro to web api\FirmWebApiDemo\FirmWebApiDemo\Authentication\BearerAuthenticationAttribute.cs:line 29
   at System.Web.Http.Filters.AuthorizationFilterAttribute.OnAuthorizationAsync(HttpActionContext actionContext, CancellationToken cancellationToken)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Filters.AuthorizationFilterAttribute.<ExecuteAuthorizationFilterAsyncCore>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Web.Http.Controllers.ExceptionFilterResult.<ExecuteAsync>d__6.MoveNext()

----------------------------------------------------------------------------------------------------
